packages/core/src/Modules/Cashier/Containers/cashier.jsx

```diff
@@ -0,0 +1,119 @@
+
+            this.props.routes.forEach(route => {
+                if ((route.path !== routes.cashier_pa || this.props.is_payment_agent_visible) &&
+                    (route.path !== routes.cashier_pa_transfer || this.props.is_payment_agent_transfer_visible)) {
+                    options.push({
+                        default: route.default,
+                        icon   : route.icon_component,
+                        label  : route.title,
+                        value  : route.component,
+                        path   : route.path,
+                    });
+                }
+            });
+
+            return options;
+        };
+
+        const action_bar_items = [
+            {
+                onClick: () => {
+                    this.props.history.push(routes.trade);
+                    this.props.toggleCashier();
+                },
+                icon : 'ModalIconClose',
+                title: localize('Close'),
+            },
+            {
+                component: () => <WalletInformation />,
+                title    : '',
+            },
+        ];
+        return (
+            <FadeWrapper
+                is_visible={this.props.is_visible}
+                className='cashier-page-wrapper'
+                keyname='cashier-page-wrapper'
+            >
+                <div className='cashier' ref={this.setWrapperRef}>
+                    <VerticalTab
+                        header_title={localize('Cashier')}
+                        action_bar={action_bar_items}
+                        action_bar_classname='cashier__inset_header'
+                        alignment='center'
+                        id='cashier'
+                        classNameHeader='cashier__tab-header'
+                        current_path={this.props.location.pathname}
+                        is_routed={true}
+                        is_full_width={true}
+                        list={menu_options()}
+                    />
+                </div>
+            </FadeWrapper>
+        );
+    }
+}
+
+Cashier.propTypes = {
+    disableRouteMode                 : PropTypes.func,
+    enableRouteMode                  : PropTypes.func,
+    history                          : PropTypes.object,
+    is_payment_agent_transfer_visible: PropTypes.bool,
+    is_payment_agent_visible         : PropTypes.bool,
+    is_visible                       : PropTypes.bool,
+    location                         : PropTypes.object,
+    onMount                          : PropTypes.func,
+    routes                           : PropTypes.arrayOf(PropTypes.object),
+    toggleCashier                    : PropTypes.func,
+};
+
+export default connect(
+    ({ modules, ui }) => ({
+        disableRouteMode        : ui.disableRouteModal,
+        enableRouteMode         : ui.setRouteModal,
+        is_visible              : ui.is_cashier_visible,
+        is_payment_agent_visible: !!(modules.cashier.config.payment_agent.filtered_list.length
```

Comment:
i think we can just take modules.cashier.config, and then use the logic to separate vars in usage

Reply:
it's not new code, it was moved from header.jsx into here. IMO it doesn't make any difference, we need to have filtered_list and agents as observables here so we can make sure once they're populated if we can have PA visible or not. so instead of importing both and checking inside, it checks it all in one go.